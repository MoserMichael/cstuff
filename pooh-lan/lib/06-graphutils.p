# test graph traversal class - graphutils.inc
include 'graphutils.inc'
include 'graphlist.inc'


r := make_graph_list()

r . addnode( ~index 1 )
r . addnode( ~index 2 )
r . addnode( ~index 3 )
r . addnode( ~index 4 )
r . addnode( ~index 5 )
r . addnode( ~index 6 )

r . addedge( ~from 1 ~to 2 )
r . addedge( ~from 1 ~to 3 )
r . addedge( ~from 2 ~to 4 )
r . addedge( ~from 2 ~to 5 )
r . addedge( ~from 5 ~to 6 )
r . addedge( ~from 5 ~to 1 )
r . addedge( ~from 2 ~to 3 )
r . addedge( ~from 3 ~to 1 )

u := make_graph_utils()

println( ~msg '*** breadth first search iterator ***' )
for i   u . breadth_first_search( ~graph r ~startnode 1 )
  println( ~msg 'node [ i ]' )
end

println( ~msg '*** depth first search iterator ***' )
for i   u . depth_first_search( ~graph r ~startnode 1 )
  println( ~msg 'node [ i ]' )
end


