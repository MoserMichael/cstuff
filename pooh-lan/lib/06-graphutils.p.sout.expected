*** breadth first search iterator ***
node 1
node 3
node 2
node 5
node 4
node 6
*** depth first search iterator ***
node 1
node 2
node 4
node 5
node 6
node 3
